.github/workflows/module-generation.yml

```diff
@@ -0,0 +1,43 @@
+name: Test Template Generators
+
+on:
+  workflow_dispatch:
+  pull_request:
+    branches:
+      - main
+      - "module-generation-cli"
+
+jobs:
+  generate_and_run_module:
+    # if: github.repository_owner == 'viamrobotics'
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: true
+      matrix:
+        resource: ["arm component"]
+    steps:
+      - name: Setup Go environment
+        uses: actions/setup-go@v5
+        with:
+          go-version: "^1.23"
+
+      - name: Checkout Code
+        uses: actions/checkout@v4
+
+      - name: Run module generator
+        run: go run ./cli/viam --debug module generate --resource "${{ matrix.resource }}"
```

Comment:
For some unknown reason, the test fails to find the generated `my-module` directory when the `--debug` flag is not used.